packages/phpstan-latte-rules/src/Rules/LatteCompleteCheckRule.php

```diff
@@ -72,35 +69,34 @@ public function __construct(
      */
     public function getNodeTypes(): array
     {
-        return [MethodCall::class];
+        return [ClassMethod::class];
     }
 
     /**
-     * @param MethodCall $node
+     * @param ClassMethod $node
      * @return RuleError[]
      */
     public function process(Node $node, Scope $scope): array
     {
-        if (! $this->templateRenderAnalyzer->isNetteTemplateRenderMethodCall($node, $scope)) {
-            return [];
-        }
-
-        $renderTemplatesWithParameters = $this->latteTemplateWithParametersMatcher->match($node, $scope);
-        $componentNamesAndTypes = $this->componentMapResolver->resolveFromMethodCall($node, $scope);
-
         $errors = [];
-        foreach ($renderTemplatesWithParameters as $renderTemplateWithParameter) {
-            $currentErrors = $this->processTemplateFilePath(
-                $renderTemplateWithParameter->getTemplateFilePath(),
-                $renderTemplateWithParameter->getParametersArray(),
-                $scope,
-                $componentNamesAndTypes,
-                $node->getLine()
-            );
-
-            $errors = array_merge($errors, $currentErrors);
+        foreach ($this->latteTemplateHolders as $latteTemplateHolder) {
+            if (! $latteTemplateHolder->check($node, $scope)) {
+                continue;
+            }
+            $renderTemplatesWithParameters = $latteTemplateHolder->findRenderTemplateWithParameters($node, $scope);
+            $componentNamesAndTypes = $latteTemplateHolder->findComponentNamesAndTypes($node, $scope);
+
+            foreach ($renderTemplatesWithParameters as $renderTemplateWithParameters) {
+                $currentErrors = $this->processTemplateFilePath(
+                    $renderTemplateWithParameters,
+                    $scope,
+                    $componentNamesAndTypes,
+                    $node->getLine()    // TODO add line to RenderTemplateWithParameters
```

Comment:
line is changed now, what do you think about adding line to RenderTemplateWithParameters value object? For controls it will be set to line where $template->render() is called and for presenters it can be line of action / render method